lib/util/strtoid.c

```diff
@@ -80,7 +80,7 @@ sudo_strtoidx_v1(const char *p, const char *sep, char **endp, const char **errst
 	 * Disallow id -1 (UINT_MAX), which means "no change"
 	 * and check for a valid separator (if specified).
 	 */
-	if (ret == (id_t)-1 || ret == (id_t)UINT_MAX || !valid_separator(p, ep, sep)) {
+	if (ret == (id_t)-1 || !valid_separator(p, ep, sep)) {
```

Comment:
This needs to stay.  On some systems id_t is a signed 64-bit integer so both -1 and UINT_MAX are possible.